openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -368,9 +374,20 @@ func updateContainersResources(containers []coreapi.Container, podAnnotations ma
 			cpuRequest := c.Resources.Requests[coreapi.ResourceCPU]
 			cpuRequestInMilli := cpuRequest.MilliValue()
 
-			cpuShares := cm.MilliCPUToShares(cpuRequestInMilli)
-			podAnnotations[cpusharesAnnotationKey] = fmt.Sprintf(`{"%s": %d}`, containerResourcesAnnotationValueKeyCPUShares, cpuShares)
+			// Casting to uint64, Linux build returns uint64, noop Darwin build returns int64
+			resourceAnno.CPUShares = uint64(cm.MilliCPUToShares(cpuRequestInMilli))
+
+			// This should not error but if something does go wrong we default to string creation of just CPU Shares
+			// and add a warning annotation
+			resourceAnnoString, err := json.Marshal(resourceAnno)
+			if err != nil {
+				podAnnotations[workloadAdmissionWarning] = fmt.Sprintf("failed to marshal cpu resources, using fallback: err: %s", err.Error())
```

Comment:
do we want to set a warning as annotation?

Reply:
Yeah wasn't 100% sure what to do here, I wanted to make it easy to identify if it cropped up in the wild, even though it's highly improbable, maybe a simple log would be good enough here, unless we feel firing off an event would be more appropriate.

Reply:
Just for clarity @ffromani , do you mean in general the use of annotations for warning in this webhook?

Reply:
I mean it feels weird to add warning data as annotation. Is the user supposed to look at annotations checking for warninga? Not sure there's a better alternative though, so this comment is not blocking.